libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerAction.cs

```diff
@@ -0,0 +1,162 @@
+            string cardNoMatchResponse = QnAMakerActionBuilder.DefaultCardNoMatchResponse, 
+            Metadata[] strictFilters = null,  
+            HttpClient httpClient = null, 
+            [CallerFilePath] string sourceFilePath = "", 
+            [CallerLineNumber] int sourceLineNumber = 0
+            )
+            : base()
+        {
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+            this.KnowledgeBaseId = knowledgeBaseId ?? throw new ArgumentNullException(nameof(knowledgeBaseId));
+            this.HostName = hostName ?? throw new ArgumentNullException(nameof(hostName));
+            this.EndpointKey = endpointKey ?? throw new ArgumentNullException(nameof(endpointKey));
+            this.Threshold = threshold;
+            this.NoAnswer = noAnswer;
+            this.ActiveLearningCardTitle = activeLearningCardTitle;
+            this.CardNoMatchText = cardNoMatchText;
+            this.CardNoMatchResponse = cardNoMatchResponse;
+            this.StrictFilters = strictFilters;
+            this.httpClient = httpClient;
+        }
+
+        [JsonConstructor]
+        public QnAMakerAction([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        [JsonProperty("knowledgeBaseId")]
+        public string KnowledgeBaseId { get; set; }
+
+        [JsonProperty("hostname")]
+        public string HostName { get; set; }
+
+        [JsonProperty("endpointKey")]
+        public string EndpointKey { get; set; }
+
+        [JsonProperty("threshold")]
+        public float Threshold { get; set; }
+
+        [JsonProperty("noAnswer")]
+        public string NoAnswer { get; set; }
+
+        [JsonProperty("activeLearningCardTitle")]
+        public string ActiveLearningCardTitle { get; set; }
+
+        [JsonProperty("cardNoMatchText")]
+        public string CardNoMatchText { get; set; }
+
+        [JsonProperty("cardNoMatchResponse")]
+        public string CardNoMatchResponse { get; set; }
+
+        [JsonProperty("strictFilters")]
+        public Metadata[] StrictFilters { get; set; }
+
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var endpoint = new QnAMakerEndpoint
+            {
+                EndpointKey = this.EndpointKey,
+                Host = this.HostName,
+                KnowledgeBaseId = this.KnowledgeBaseId
+            };
+
+            var qnamakerOptions = new QnAMakerOptions
+            {
+                ScoreThreshold = this.Threshold,
+                StrictFilters = this.StrictFilters
+            };
+
+            if (qnamaker == null)
+            {
+                qnamaker = new QnAMaker(endpoint, qnamakerOptions, httpClient);
```

Comment:
Just double checking: With the default constructor, httpClient is null. Does this code work in that case?

Reply:
Yes, code works without httpClient